src/doc_builder/commands/preview.py

```diff
@@ -0,0 +1,225 @@
+            If so, returns mdx file path.
+            """
+            if datetime.now() - self.last_modified < timedelta(seconds=1):
+                return False, None, None
+
+            self.last_modified = datetime.now()
+            src_path = event.src_path
+            relative_path = event.src_path[len(self.args.path_to_docs) + 1 :]
+            is_valid_file = False
+            if not event.is_directory:
+                if src_path.endswith(".py") and src_path in self.source_files_mapping:
+                    src_path = self.source_files_mapping[src_path]
+                if src_path.endswith(".mdx"):
+                    is_valid_file = True
+                    return is_valid_file, src_path, relative_path
+            return is_valid_file, src_path, relative_path
+
+        def build(self, src_path, relative_path):
+            """
+            Build single mdx file in a temp dir.
+            """
+            print(f"Building: {src_path}")
+            # copy the built files into the actual build folder dawg
+            with tempfile.TemporaryDirectory() as tmp_input_dir:
+                # copy the file into srcpath
+                shutil.copy(src_path, tmp_input_dir)
+
+                with tempfile.TemporaryDirectory() as tmp_out_dir:
+                    build_doc(
+                        "datasets",
+                        tmp_input_dir,
+                        tmp_out_dir,
+                        version=self.args.version,
+                        language=self.args.language,
+                        watch_mode=True,
+                    )
+                    src = Path(tmp_out_dir) / Path(src_path).name
+                    dest = self.kit_routes_folder / relative_path
+                    shutil.move(src, dest)
+
+        def start(self):
+            """
+            Starts `pywatchdog.observer` for listening file changes.
+            """
+            observer = Observer()
+            observer.schedule(self, self.path, recursive=True)
+            observer.start()
+            print(f"\nWatching for changes in: {self.path}\n")
+            try:
+                while True:
+                    time.sleep(1)
+            except KeyboardInterrupt:
+                observer.stop()
+            observer.join()
+
+
+def start_sveltekit_dev(tmp_dir, env, args):
+    """
+    Installs sveltekit node dependencies & starts sveltekit in dev mode in a temp dir.
+    """
+    working_dir = str(tmp_dir / "kit")
+    print("Installing node dependencies")
+    subprocess.run(
+        ["npm", "ci"],
+        stdout=subprocess.PIPE,
+        check=True,
+        encoding="utf-8",
+        cwd=working_dir,
+    )
+
+    # start sveltekit in dev mode
+    subprocess.run(
+        ["npm", "run", "dev"],
+        check=True,
+        encoding="utf-8",
+        cwd=working_dir,
+        env=env,
+    )
+
+
+def preview_command(args):
+    if is_watchdog_available():
```

Comment:
To avoid everything being indented here, I'd test the negation and raise the `ImportError`, then just unindent the whole function.

Reply:
https://github.com/huggingface/doc-builder/pull/139/commits/979e8d981a887d911374d9d5e94fdf2b1e22a0ad